include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -254,3 +272,24 @@ class alignment_result
 };
 
 } // namespace seqan3
+
+namespace seqan3::detail
+{
+/*!\brief Transformation trait to access the hidden result value type of the seqan3::alignment_result class.
+ * \implements transformation_trait
+ * \ingroup pairwise_alignment
+ *
+ * \tparam result_t The type of the alignment result.
+ *
+ * \details
+ *
+ * The template parameter must be a specialisation of seqan3::alignment_result otherwise this class is incomplete.
+ */
+template <typename result_value_t>
+struct alignment_result_value_type_accessor<alignment_result<result_value_t>>
```

Comment:
I thought you wanted to befriend the algorithm anyway, can't you do that and then access a private member type of the class?

Reply:
After todays meeting is this now resolved?